x-pack/plugins/enterprise_search/common/types/app_search.ts

```diff
@@ -30,3 +30,31 @@ export interface IConfiguredLimits {
     maxEnginesPerMetaEngine: number;
   };
 }
+
+export interface IApiToken {
+  access_all_engines?: boolean;
```

Comment:
This is not a blocker/me stating a personal preference, but it would be nice if we didn't jump between `snake_cased` and `camelCased` keys, at least in the client-side. Ideally we'd use our server API endpoints to transform snake_cased data into camelCased, but I understand that's a nice-to-have/tech debt item and am fine deferring that for later.

Reply:
After some reflection I think it's probably a pipe dream for us to ever lose incidental snake_cased data in our codebase, at least not before 9.x 😛 I'll collapse this comment as useless, haha